src/prefect/_internal/concurrency/calls.py

```diff
@@ -236,6 +236,37 @@ class Call(Generic[T]):
     timeout: Optional[float]
     runner: Optional["Portal"] = None
 
+    def __eq__(self, other: object) -> bool:
+        if self is other:
+            return True
+        if not isinstance(other, Call):
+            return NotImplemented
```

Comment:
why this instead of `False`? jw

Reply:
my thinking was that returning `NotImplemented` when `other` is not a `Call` instance allows `other` a chance to define how it compares to a `Call` object (via its own `__eq__` method). If it returned False directly, it would unilaterally decide they are not equal, preventing the other object from having any say in the comparison. like for example, if there were `SomeCallWrapper` that defines its own `__eq__`